mlir-tensorrt/compiler/lib/Dialect/Plan/Transforms/RefineTypes.cpp

```diff
@@ -294,6 +295,30 @@ struct StableHloRefineTypeFromWithShapeGeneric
   }
 };
 
+struct TensorRTResizeAbsorbCastPattern: public OpRewritePattern<tensor::CastOp> {
+  using OpRewritePattern<tensor::CastOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(tensor::CastOp castOp,
+                                PatternRewriter &rewriter) const override {
+    auto resizeOp = castOp.getOperand().getDefiningOp();
+
+    // tensor.cast is the only user of the tensorrt op
+    // tensor.cast must be a refining op
+    if (!resizeOp || !resizeOp->hasOneUse() ||
+        !isa<tensorrt::TensorRTDialect>(resizeOp->getDialect()) || 
+        !tensorrt::isTargetRefinementOfSource(
+            castOp.getSource().getType().getShape(),
+            castOp.getType().getShape()))
+      return failure();
+    
+    // how to rewrite?
```

Comment:
This is the correct way to modify the type of resize op. But you also just need (after line 317)

```
rewriter.replaceOp(castOp, rewsizeOp);
```

since the `tensor.cast` is now unnecessary